spotify_player/src/cli/client.rs

```diff
@@ -350,3 +362,359 @@ async fn handle_playback_request(
+        Context::Playlist { tracks, playlist } => (
+            tracks.into_iter().map(|t| TrackData {
+                id: t.id,
+                name: t.name,
+            }),
+            playlist.name,
+        ),
+        _ => unreachable!(),
+    };
+
+    // Get import dir/file
+    let cache_dir = get_cache_folder_path()?;
+    let imports_dir = cache_dir.join("imports");
+    let to_dir = imports_dir.join(import_to.id());
+    let from_file = to_dir.join(import_from.id());
+
+    if !to_dir.exists() {
+        create_dir_all(to_dir)?;
+    }
+    // Construct hash sets of the playlists' track IDs
+    let to_hash_set: HashSet<TrackData> = HashSet::from_iter(to_tracks);
+    let from_hash_set: HashSet<TrackData> = HashSet::from_iter(from_tracks);
+
+    let mut new_tracks_hash_set = &from_hash_set - &to_hash_set;
+
+    let mut result = String::new();
+    result += &format!(
+        "Importing tracks from {}:{} to {}:{}...\n",
+        import_from.id(),
+        from_name,
+        import_to.id(),
+        to_name
+    );
+
+    let mut track_buff = Vec::new();
+    if from_file.exists() {
+        let hash_set_bytes = std::fs::read(&from_file).context(format!(
+            "Read cached playlist import data from {}",
+            from_file.display()
+        ))?;
+        std::fs::remove_file(&from_file)?;
+        let old_from_hash_set: HashSet<TrackData> =
+            serde_json::from_slice(&hash_set_bytes).context("Deserialize playlist import data")?;
+
+        // Only consider new tracks that were not included in the previous import.
+        new_tracks_hash_set = &new_tracks_hash_set - &old_from_hash_set;
+
+        // If `delete` option is specified, delete previously imported tracks that are not in the current `from` playlist
+        let deleted_hash_set = &old_from_hash_set - &from_hash_set;
+        if delete {
+            result +=
+                &format!("Deleting previously imported tracks in {to_name} that are not in the current {from_name}...\n");
+
+            for t in &deleted_hash_set {
+                track_buff.push(PlayableId::Track(t.id.as_ref()));
+
+                if track_buff.len() == TRACK_BUFFER_CAP {
+                    client
+                        .spotify
+                        .playlist_remove_all_occurrences_of_items(
+                            import_to.as_ref(),
+                            track_buff,
+                            None,
+                        )
+                        .await?;
+                    track_buff = Vec::new();
+                }
+            }
+
+            if !track_buff.is_empty() {
+                client
+                    .spotify
+                    .playlist_remove_all_occurrences_of_items(import_to.as_ref(), track_buff, None)
+                    .await?;
+            }
+            result += &format!("Deleted tracks: \n");
+        } else {
+            result += &format!("Import removed tracks: \n");
+        }
```

Comment:
Why do you need to handle this case (no `delete` option is specified)?

Reply:
I thought it may be confusing to say "Deleted tracks:" when it they were deleted from the imported playlist and not the import playlist.

Reply:
I mean you shouldn't handle this case. If no `delete` option is specified, it should not print anything, even information about deleted tracks from the imported playlist.